seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceReader.java

```diff
@@ -101,9 +103,13 @@ public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
             for (TopicPartition partition : partitions) {
                 for (ConsumerRecord<byte[], byte[]> record : records.records(partition)) {
 
-                    String v = stringDeserializer.deserialize(partition.topic(), record.value());
-                    String t = partition.topic();
-                    output.collect(new SeaTunnelRow(new Object[]{t, v}));
+                    if(deserializationSchema != null){
+                        deserializationSchema.deserialize(record.value(),output);
+                    }else{
+                        String v = stringDeserializer.deserialize(partition.topic(), record.value());
+                        String t = partition.topic();
+                        output.collect((T) new SeaTunnelRow(new Object[]{t, v}));
```

Comment:
If user does not assign schema, it will generate a simple schema as the following shown:

```
content
data
```
So the row only have one field named `content`, but you generate a row with two fields.